File: src/main/scala/com/wangheng/day05/TextLength.py
from pyspark import SparkContext

if __name__ == '__main__':
    sc = SparkContext("local[*]", "Text Length")
    lines = sc.textFile("hdfs://localhost:9000/spark_test_data.txt", 1)
    line_count = lines.map(lambda x:len(x))
    count = line_count.reduce(lambda x, y: x+y)
    print("file's length is : ", count)

File: src/main/scala/com/wangheng/core/day05/TextLength.java
package com.wangheng.core.day05;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.Function2;

public class TextLength {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf().setAppName("text_length").setMaster("local[*]");
        JavaSparkContext sc = new JavaSparkContext(conf);
        JavaRDD<String> lines = sc.textFile("hdfs://localhost:9000/spark_test_data.txt", 1);
        JavaRDD<Integer> lineLength = lines.map(new Function<String, Integer>() {
            @Override
            public Integer call(String s) throws Exception {
                return s.length();
            }
        });
        int count = lineLength.reduce(new Function2<Integer, Integer, Integer>() {
            @Override
            public Integer call(Integer integer, Integer integer2) throws Exception {
                return integer + integer2;
            }
        });
        System.out.println("file's length is :" + count);
        sc.close();
    }
}
